common/include/nbl/examples/common/BuiltinResourcesApplication.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2023-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_EXAMPLES_BUILTIN_RESOURCE_APPLICATION_HPP_INCLUDED_
+#define _NBL_EXAMPLES_BUILTIN_RESOURCE_APPLICATION_HPP_INCLUDED_
+
+
+// we need a system, logger and an asset manager
+#include "nbl/application_templates/MonoAssetManagerApplication.hpp"
+
+#ifdef NBL_EMBED_BUILTIN_RESOURCES
+// TODO: the include/header `nbl/examples` archive
+// TODO: the source `nbl/examples` archive
+// TODO: the build `nbl/examples` archive
+// TODO: make the `this_example` optional, only if the example has builtins
+#include "nbl/this_example/builtin/CArchive.h"
```

Comment:
@YasInvolved I think this can be done with `__has_include` https://en.cppreference.com/w/cpp/preprocessor/include